src/umpire/ResourceManager.cpp

```diff
@@ -444,6 +444,36 @@ void ResourceManager::memset(void* ptr, int value, std::size_t length)
   op->apply(ptr, alloc_record, value, length);
 }
 
+camp::resources::Event ResourceManager::memset(void* ptr, int value, camp::resources::Resource& ctx, std::size_t length)
+{
+  UMPIRE_LOG(Debug, "(ptr=" << ptr << ", value=" << value << ", length=" << length << ")");
+
+  auto& op_registry = op::MemoryOperationRegistry::getInstance();
+
+  auto alloc_record = m_allocations.find(ptr);
+
+  std::ptrdiff_t offset = static_cast<char*>(ptr) - static_cast<char*>(alloc_record->ptr);
+  std::size_t size = alloc_record->size - offset;
+
+  if (length == 0) {
+    length = size;
+  }
+
+  UMPIRE_REPLAY(R"( "event": "memset", "payload": { )"
```

Comment:
Side node: While recording these events is perhaps important, we should have a review about whether replay of these kinds of operations are appropriate.  The replay tool currently does not fully support operations on memory that is already allocated (other than `realloc`).